packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1585,6 +1585,11 @@ abstract class BuildContext {
   /// (directly or indirectly) from build methods, layout and paint callbacks, or
   /// from [State.didChangeDependencies].
   ///
+  /// This method should not be called from [State.deactivate] or [State.dispose]
+  /// because the element tree is no longer stable at that time. To refer to
+  /// an ancestor from one of those methods, save a reference to the ancestor
+  /// in [State.initState] and update the reference in [State.didChangeDependencies].
```

Comment:
It'll assert if you call this in initState. You should just do it in didChangeDependencies.